src/cpu/binary_injector_utils.cpp

```diff
@@ -34,7 +34,7 @@ std::vector<const void *> prepare_binary_args(const post_ops_t &post_ops,
             post_ops_binary_rhs_arg_vec.emplace_back(CTX_IN_MEM(const void *,
                     DNNL_ARG_ATTR_MULTIPLE_POST_OP(idx) | DNNL_ARG_SRC_1));
         }
-#if DNNL_X64
+#if defined(DNNL_X64)
```

Comment:
I'm not sure if it's correct. All arch specific macros are defined but only one of them must be defined as `1` and the others must be defined as `0`.

See https://github.com/uxlfoundation/oneDNN/blob/96be1ead1444c328e1c0351bec88c18370639a3e/src/cpu/platform.hpp#L34 for more details.

Reply:
Let me double-check when did it report this hit and for which backend.
But it definitely was highlighted. It just means that it's not compiled with platform.hpp at least.

Reply:
I believe there is inconsistency in how we identify the architecture. We have two ways to do that:
1. Through CMake: https://github.com/uxlfoundation/oneDNN/blob/96be1ead1444c328e1c0351bec88c18370639a3e/CMakeLists.txt#L64-L76 and here https://github.com/uxlfoundation/oneDNN/blob/96be1ead1444c328e1c0351bec88c18370639a3e/cmake/platform.cmake#L32
2. Through the platform.hpp: https://github.com/uxlfoundation/oneDNN/blob/96be1ead1444c328e1c0351bec88c18370639a3e/src/cpu/platform.hpp#L40-L53

So, it seems that when it's identified via CMake then there is only macro defined but when it's identified via platform.hpp then all macros are defined.

So I think we should make it consistent before starting to rely on the identification logic.

Reply:
FYI: MFDNN-5545

Reply:
So, I'm definitely not covering build and aligning macro definition change as a part of this PR.

I dropped the check for X64 from this file, will monitor if Aarch64 backend fails anywhere. And if yes, will strip support from the implementation level at runtime which must have been done in the first place.

Reply:
My point is that the code `#if defined(DNNL_ANYARCH)` is not correct based on our own specification of the `DNNL_ANYARCH` macros because they are always defined if the `platform.hpp` header is included. The behavior of `#if defined(DNNL_ANYARCH)` code depends on whether `platform.hpp` was included directly or indirectly.

Reply:
DNNL_AARCH64_USE_ACL doesn't represent an architecture, so I'm not sure what your comment refers to. It is the way it was initially designed. If the intention to make it acting like architecture, then it's a different kind of project not related to Wundef issues.

Reply:
Never mind, I confused DNNL_AARCH64_USE_ACL with DNNL_AARCH64.